src/SSEConnection.js

```diff
@@ -0,0 +1,110 @@
+import AbstractConnection from "./AbstractConnection";
+import ConnectionEvent from "./ConnectionEvent";
+import { List } from "immutable";
+
+/**
+ * Server Sent Event Connections
+ */
+export default class SSEConnection extends AbstractConnection {
+  /**
+   * Initializes an Instance of SSEConnection
+   * @constructor
+   * @param {string} url – URL to connect to
+   * @param {object} [options]
+   * @param {boolean} [options.withCredentials=true] – used method
+   */
+  constructor(url, options = {}) {
+    super(url);
+
+    Object.defineProperty(this, "options", {
+      value: { withCredentials: options.withCredentials || false }
+    });
+  }
+
+  /**
+   * callback for server sent events
+   *
+   * @callback eventCallback
+   * @param {object} [event]
+   * @param {string} [event.data]
+   */
+
+  /**
+   * Adds an event listener for the type
+   * @param {string} type
+   * @param {eventCallback} callback
+   */
+  addEventListener(type, callback) {
+    const addListenerToSource = () =>
+      this.source.addEventListener(type, callback);
+
+    if (this.state === SSEConnection.OPEN) {
+      addListenerToSource();
+    } else {
+      // not yet started, so defer the initialization
+      this.on(ConnectionEvent.OPEN, addListenerToSource);
```

Comment:
Could it be that we miss some events because we add listeners _after_ we open a connection? Why do we need deferring them?

Reply:
This is for the event listeners that are directly on the SSE source. We can not bind anything on the source before it has started and AFAIK we can not create the source without automatically starting it.